src/commands/preload/index.ts

```diff
@@ -161,6 +163,40 @@ Can be repeated to add multiple certificates.\
 			);
 		}
 
+		// Verify that image is not enabled for secure boot. First, confirm it
+		// is a secure boot image with an /opt/*.sig file in the rootA partition.
+		const { explorePartition, BalenaPartition } = await import(
+			'../../utils/image-contents'
+		);
+		const isSecureBoot = await explorePartition<boolean>(
+			params.image,
+			BalenaPartition.ROOTA,
+			async (fs: typeof Fs): Promise<boolean> => {
+				try {
+					const promiseDir = promisify(fs.readdir);
+					const files = await promiseDir('/opt');
+					return files.some((el) => el.endsWith('balenaos-img.sig'));
+				} catch {
+					// Typically one of:
+					// - Error: No such file or directory
+					// - Error: Unsupported filesystem.
+					// - ErrnoException: node_ext2fs_open ENOENT (44) args: [5261576,5268064,"r",0]
+					return false;
+				}
+				return false;
+			},
+		);
+		// Next verify that config.json enables secureboot.
+		if (isSecureBoot) {
```

Comment:
Not relevant to this PR, but maybe elsewhere we can add a check to the CLI that detects if someone has tried to enable secureboot on a non signed image, and warn them that this won't work, to avoid any confusion / chance of them thinking they've deployed a secure device but haven't!

Reply:
I see the CLI already has a [utility function](https://github.com/balena-io/balena-cli/blob/4f665f43d2a91e2e4a5b4c2a98840424e6a0bc3c/src/utils/config.ts#L181) for such a check for `os configure`. Is that sufficient? 

In that more generic case, is it better to check the image itself as we do here? Actually AIUI there are older versions of the secure boot implementation that do not require a signature file, so a server check is appropriate for that case, correct?